disassemblers/ofrak_angr/ofrak_angr/components/blocks/unpackers.py

```diff
@@ -226,3 +238,43 @@ def _angr_get_dword_blocks(
             xref = get_real_address_if_arm(angr_analysis.project.arch, xref) if xref else None
 
             yield DataWord(cb_data_xref_addr, word_size, format_string, (xref,))
+
+
+def _get_bb_exit_addr_info(
+    angr_analysis,
+    angr_complex_block,
+    angr_cb_basic_blocks,
+    current_angr_bb,
+    current_bb_idx,
+) -> Tuple[bool, Optional[int]]:
+    ## Fetch the exit point addr (if it exists) and sanity check the selection
+    if current_angr_bb.codenode in angr_complex_block.endpoints:
+        return True, None
+
+    if current_bb_idx == len(angr_cb_basic_blocks) - 1:
+        LOGGER.error(
+            f"Exit point defined for BB 0x{current_angr_bb.addr:x} even though it is the last BB on the addr list"
+        )
+        raise UnpackerError()
+
+    # If no conditional branches taken, execution "falls through" to next basic block
+    fallthrough_vaddr = get_real_address_if_arm(
+        angr_analysis.project.arch, angr_cb_basic_blocks[current_bb_idx + 1].addr
+    )
+    successors_generator = angr_complex_block.graph.succ.get(current_angr_bb.codenode)
+
+    if successors_generator is None:
+        LOGGER.warning(
+            f"angr did not find any successors for BB 0x{current_angr_bb.addr:x}, but since it has a BB after it, assuming that it still can fallthrough to the next BB."
+        )
+        return True, fallthrough_vaddr
+
+    successor_vaddrs = [
+        get_real_address_if_arm(angr_analysis.project.arch, succ_codenode.addr)
+        for succ_codenode, edge_info in successors_generator.items()
+    ]
+
+    if fallthrough_vaddr in successor_vaddrs:
+        return False, fallthrough_vaddr
+    else:
+        return False, successor_vaddrs[0]
```

Comment:
If the fallthrough (which is the next basic block) isn't a successor, it means that the basic block can't actually exit to the next basic block. For example, the last instruction is an unconditional branch (and not a linking branch, i.e. a call that returns to the same place). So it just checks if the fallthrough address is actually a successor, if it is, then it should always be the successor. Otherwise, it chooses the first successor, which is a bit arbitrary, but is probably the only successor (in the unconditional branch case) and if it isn't, then the control flow is "weird" (possibly jump table).

Actually, is that the expected behavior when the BB doesn't fall through?